worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java

```diff
@@ -961,10 +961,11 @@ public int center(Region region, Pattern pattern) throws MaxChangedBlocksExcepti
         checkNotNull(pattern);
 
         Vector center = region.getCenter();
+        Vector vec2 = center.toBlockVector();
         Region centerRegion = new CuboidRegion(
                 getWorld(), // Causes clamping of Y range
                 new Vector((int) center.getX(), (int) center.getY(), (int) center.getZ()),
-                center.toBlockVector());
+                new Vector(Math.signum(vec2.getX())*Math.round(Math.abs(vec2.getX())), vec2.getY(), Math.signum(vec2.getZ())*Math.round(Math.abs(vec2.getZ()))));
```

Comment:
I'd still like this to be pulled out to its own method to reduce the noise on this line.